ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -26,7 +26,7 @@ const APPROVE_TIMEOUT = MILLISECOND * 1200;
 
 function getDefaultSelectedAccounts(currentAddress, permissionsRequest) {
   const permission =
-    permissionsRequest.permissions?.[RestrictedMethods.eth_accounts];
+    permissionsRequest?.permissions?.[RestrictedMethods.eth_accounts];
```

Comment:
Why is this change needed?

Reply:
I saw some crashes in E2E due to this, I think it's a race condition we hadn't considered.

Unrelated to the other changes in this PR, but seemed worth fixing.